### PR TITLE
UIEH-1185: Add tests for NoteCreate, NoteEdit, NoteView Routes

### PR DESCRIPTION
**Pull request by Svitlana-Korol**
https://issues.folio.org/browse/UIEH-1185

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,8 @@
 * Add tests for ApplicationRoute component. (UIEH-1184)
 * Add tests for TitlesTable component. (UIEH-1181)
 * Add missing permissions to `package.json`. (UIEH-1222)
+* Add tests for NoteCreate, NoteEdit, NoteView Routes. (UIEH-1185)
+
```
Comment (Vladyslav-Velytskyi) [inline CHANGELOG.md:17]: Please, remove this unnecessary empty string

File: src/routes/note-edit.test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import { MemoryRouter } from "react-router-dom";
+import { render, cleanup, act, fireEvent } from '@testing-library/react';
```
Comment (Vladyslav-Velytskyi) [inline src/routes/note-edit.test.js:2]: When we destructuring more then one element from the file, we write each variable on separate string, please, refactor it.

```suggestion
import {
  render,
  cleanup,
  act,
  fireEvent,
} from '@testing-library/react';
```

File: src/routes/note-create.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import Harness from '../../test/jest/helpers/harness';
+
+const location = {
+  pathname: 'pathname',
+  search: '',
+  hash: '',
+};
+
+const history = {
+  goBack: jest.fn()
```
Comment (Vladyslav-Velytskyi) [inline src/routes/note-create.test.js:14]: Don't forget about commas after each value in object
```suggestion
  goBack: jest.fn(),
```

File: src/routes/note-create.test.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    history.goBack.mockClear();
+  });
+
+  afterEach(cleanup);
+
+  it('should render NoteCreateRoute', async () => {
+    await act(async () => {
+      await renderNoteCreateRoute();
+    });
+  });
```
Comment (Vladyslav-Velytskyi) [inline src/routes/note-create.test.js:42]: What are you testing here? It should be `expect` construction inside `it`'s callback

File: src/routes/note-edit.test.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        history.goBack.mockClear();
+    });
+
+    afterEach(cleanup);
+
+    it('should render NoteEditRoute', async () => {
+        await act(async () => {
+            await renderNoteEditRoute();
+        });
+    });
```
Comment (Vladyslav-Velytskyi) [inline src/routes/note-edit.test.js:65]: This is not a test too

File: src/routes/note-view.test.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    history.push.mockClear();
+  });
+
+  afterEach(cleanup);
+
+  it('should render NoteViewRoute', async () => {
+    await act(async () => {
+      await renderNoteViewRoute();
+    });
+  });
```
Comment (Vladyslav-Velytskyi) [inline src/routes/note-view.test.js:75]: Rewrite it please too

File: src/routes/note-edit.test.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+    describe('when navigateBack click', () => {
+        it('should navigateBack', () => {
```
Comment (Vladyslav-Velytskyi) [inline src/routes/note-edit.test.js:68]: Suggestion to rename
```suggestion
    describe('when click on navigate back button', () => {
        it('should handle "history.goBack"', () => {
```

**Review by Vladyslav-Velytskyi — CHANGES_REQUESTED** [review]
Please, also fix lint issues, related to your changes

File: src/routes/note-create.test.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import { MemoryRouter } from "react-router-dom";
+import { render, cleanup, act } from '@testing-library/react';
```
Comment (mariiaschv) [inline src/routes/note-create.test.js:2]: Please, refactor this also to have variables on separate lines.

File: src/routes/note-edit.test.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+import { render, cleanup, act, fireEvent } from '@testing-library/react';
+
+import NoteEditRoute from './note-edit';
+import Harness from '../../test/jest/helpers/harness';
+
+jest.mock('@folio/stripes/smart-components', () => ({
+    NoteEditPage:
+        ({
+            navigateBack,
+        }) => (
```
Comment (mariiaschv) [inline src/routes/note-edit.test.js:11]: I think, you can leave it on one line here as you have only one variable here.
```suggestion
        ({ navigateBack }) => (
```

File: src/routes/note-view.test.js [hunk 10]
```diff
@@ -0,0 +1,96 @@
+import { MemoryRouter } from "react-router-dom";
+import { render, cleanup, act, fireEvent } from '@testing-library/react';
```
Comment (mariiaschv) [inline src/routes/note-view.test.js:2]: This also should be on separate lines.

File: src/routes/note-view.test.js [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+  afterEach(cleanup);
+
+  it('should render NoteViewRoute', async () => {
+    await act(async () => {
+      await renderNoteViewRoute();
+    });
+  });
+
+  describe('when when click on navigateBack button', () => {
+    it('should navigate back', () => {
```
Comment (mariiaschv) [inline src/routes/note-view.test.js:78]: You test here a case when you don't have `location.state`, so I suggest to rename the test case:
```suggestion
    it('should redirect to eholdings page', () => {
```

But you also have a case with `location.state` present and there it should handle "history.goBack" and you can add a test for it too.

**Review by mariiaschv — CHANGES_REQUESTED** [review]

**Review by mariiaschv — COMMENTED** [review]
And you also should sigh the required Contributor License Agreement to have the checks passed.

File: src/routes/note-edit.test.js [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+  fireEvent, 
+} from '@testing-library/react';
+
+import NoteEditRoute from './note-edit';
+import Harness from '../../test/jest/helpers/harness';
+
+jest.mock('@folio/stripes/smart-components', () => ({
+    NoteEditPage:
+        ({ navigateBack }) => (
+            <>
```
Comment (BogdanDenis) [inline src/routes/note-edit.test.js:14]: We can simply return `<button>` without `<>` because it's a single element

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

**Review by BogdanDenis — APPROVED** [review]

**Review by mariiaschv — APPROVED** [review]

**Review by Vladyslav-Velytskyi — APPROVED** [review]

**Pull request merged**